<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="world_file"  default="$(find comp_cov_sim)/custom_worlds/empty.world"/>
	<arg name="base"      value="$(optenv TURTLEBOT_BASE kobuki)"/> <!-- create, roomba -->
	<arg name="battery"   value="$(optenv TURTLEBOT_BATTERY /proc/acpi/battery/BAT0)"/>  <!-- /proc/acpi/battery/BAT0 --> 
	<arg name="gui" default="false"/>
	<arg name="stacks"    value="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons --> 
	<arg name="3d_sensor" value="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro --> 
	<arg name="map_file" value="$(find comp_cov_sim)/custom_maps/map.yaml"/>
	<arg name="robot_0_name" default="robot_0"/>
	<arg name="robot_1_name" default="robot_1"/>
	<arg name="0_x" default="0.0"/>
	<arg name="0_y" default="0.0"/>
	<arg name="1_x" default="62.0"/>
	<arg name="1_y" default="62.0"/>
	<arg name="seed" default="0"/>
	<arg name="accelerate_time" default="100"/>
  

  <group>
		<include file="$(find gazebo_ros)/launch/empty_world.launch">
			<arg name="use_sim_time" value="true"/>
			<arg name="debug" value="false"/>
			<arg name="gui" value="$(arg gui)" />
			<arg name="world_name" value="$(arg world_file)"/>
			<arg name="use_clock_frequency" value="true"/>
			<arg name="pub_clock_frequency" value="$(arg accelerate_time)"/>
		</include>

		<group ns="$(arg robot_0_name)_ns">
			<param name="tf_prefix" value="$(arg robot_0_name)"/>
			<include file="$(find comp_cov_sim)/launch/includes/kobuki.launch.xml">
				<arg name="base" value="$(arg base)"/>
				<arg name="stacks" value="$(arg stacks)"/>
				<arg name="3d_sensor" value="$(arg 3d_sensor)"/>
				<arg name="initial_pose" value="-x $(arg 0_x) -y $(arg 0_y)"/>
				<arg name="robot_name" value="$(arg robot_0_name)"/>
			</include>

			<!-- map node -->
			<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>

			<node name="tf" pkg="tf" type="static_transform_publisher" args="$(arg 0_x) $(arg 0_y) 0 0 0 0 /map /$(arg robot_0_name)/odom 100"/>

			<!-- Launch coverage node -->
			
			<node name="coverage_node" pkg="comp_cov_sim" type="coverage_node.py" output="screen" required="True">
				<param name="robot_size" value="0.35"/>
				<param name="robot_name" value="$(arg robot_0_name)"/>
				<param name="init_pos" value="$(arg 0_x) $(arg 0_y)"/>
				<param name="is_manager" value="True"/>
			</node>
		</group>
 </group>

  <group ns="$(arg robot_1_name)_ns">
	<param name="tf_prefix" value="$(arg robot_1_name)"/>
 	  <include file="$(find comp_cov_sim)/launch/includes/kobuki.launch.xml">
	    	<arg name="base" value="$(arg base)"/>
   		<arg name="stacks" value="$(arg stacks)"/>
		<arg name="3d_sensor" value="$(arg 3d_sensor)"/>
		<arg name="initial_pose" value="-x $(arg 1_x) -y $(arg 1_y)"/>
    		<arg name="robot_name" value="$(arg robot_1_name)"/>
 	 </include>

	
	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>
	
	<node name="tf" pkg="tf" type="static_transform_publisher" args="$(arg 1_x) $(arg 1_y) 0 0 0 0 /map $(arg robot_1_name)/odom 100" />

	<node name="coverage_node" pkg="comp_cov_sim" type="coverage_node.py" output="screen">
		<param name="robot_size" value="0.35"/>
		<param name="robot_name" value="$(arg robot_1_name)"/>
		<param name="init_pos" value="$(arg 1_x) $(arg 1_y)"/>
		<param name="is_manager" value="False"/>
		<param name="seed" value="$(arg seed)"/>
	</node>
  </group>

  <!-- WHY DO WE NEED THIS?
  <node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
        args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
    <param name="scan_height" value="10"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.45"/>
    <remap from="image" to="/camera/depth/image_raw"/>
    <remap from="scan" to="/scan"/>
  </node>
-->
</launch>
